### PR TITLE
#fix: 107491

### LINKED ISSUE #107491: Selecting 2 color values and changing them both with color picker
As the title says, it would be a very nice feature if you could select 2 identical color values (HEX, RGB or HSL) and change them both at once with the color picker.

The value does not change for the 2nd selected option when changing colors

j3XxbaQ4Ux

### PR DESCRIPTION
**Pull request by orange4glace**
This PR fixes #107491 

### Behavior
- Select multiple colors and changing color in ColorPicker changes every selected colors with respecting its color representation.
- Select multiple colors and changing color which is not selected will changes hovered color but will not changes selected colors.
- Non-color selections will be ignored.

Honeycam 2020-10-15 23-16-43

--------------------------

Honeycam 2020-10-15 23-17-15

-------------------------

Honeycam 2020-10-15 23-19-19

### CODE REVIEW

**Review by thunder-coding — APPROVED** [review]

File: src/vs/editor/contrib/hover/modesContentHover.ts [hunk 1]
```diff
@@ -379,72 +379,123 @@ export class ModesContentHoverWidget extends ContentHoverWidget {
...
 							color: {
 								red: color.rgba.r / 255,
 								green: color.rgba.g / 255,
 								blue: color.rgba.b / 255,
 								alpha: color.rgba.a
 							}
-						}, msg.provider, CancellationToken.None).then((colorPresentations) => {
-							model.colorPresentations = colorPresentations || [];
+						}, msg.provider, CancellationToken.None))).then((colorPresentationsList) => {
+							for (let i = 0; i < models.length; i++) {
```
Comment (NotWearingPants) [inline src/vs/editor/contrib/hover/modesContentHover.ts:486]: `for (const model of models)`

File: src/vs/editor/contrib/hover/modesContentHover.ts [hunk 2]
```diff
@@ -379,72 +379,123 @@ export class ModesContentHoverWidget extends ContentHoverWidget {
...
 
 				const editorModel = this._editor.getModel();
-				let range = new Range(msg.range.startLineNumber, msg.range.startColumn, msg.range.endLineNumber, msg.range.endColumn);
-				let colorInfo = { range: msg.range, color: msg.color };
 
-				// create blank olor picker model and widget first to ensure it's positioned correctly.
-				const model = new ColorPickerModel(color, [], 0);
-				const widget = new ColorPickerWidget(fragment, model, this._editor.getOption(EditorOption.pixelRatio), this._themeService);
+				let targetIndex = 0;
+				let models: ColorPickerModel[] = [];
```
Comment (NotWearingPants) [inline src/vs/editor/contrib/hover/modesContentHover.ts:384]: All of these arrays can be `const`

**Comment by aiday-mar** [comment]
Hi @orange4glace thank you for your PR. In order for this PR to be considered for a merge into VS Code, could you resolve the merge conflict and address the review comments for polishing the code? Thank you.

**Comment by aiday-mar** [comment]
Hi @orange4glace, due to the inactivity on this PR, it will be closed. Thank you for your understanding.

**Pull request closed without merging**